src/main/java/com/amazonaws/encryptionsdk/ParsedCiphertext.java

```diff
@@ -33,9 +35,12 @@
      * {@code ciphertext} and that any changes made to the backing array will be reflected here as
      * well.
      */
-    public ParsedCiphertext(final byte[] ciphertext) {
+    public ParsedCiphertext(final byte[] ciphertext) throws ParseException {
```

Comment:
Because the constructor for ParsedCiphertext can now throw an error, we probably want to check everywhere ParsedCiphertext is being constructed and verify that the error will be handled appropriately, and make sure we have a test to ensure that it continues to handle in the same way. For example, we currently test the happy case for using encrypt/decrypt with ParsedCiphertext: https://github.com/aws/aws-encryption-sdk-java/blob/master/src/test/java/com/amazonaws/encryptionsdk/AwsCryptoTest.java#L199

This PR should include a new test in AwsCryptoTest to test the correct behavior of encrypt/decrypt in the case that ParsedCiphertext throws this error during construction.

Reply:
While this is unclear, I'd probably through a `BadCiphertextException` rather than a `ParseException`. This is because the latter appears to be used more for low-level primitive parsing rather than trying to understand an entire ciphertext.

Regardless, these are both descendants of `RuntimeException` and thus need-not (and likely should-not) be declared in a `throws` statement on the method.

Either way, more testing is in order.